### PR TITLE
4035 Add user registration processing

### PR DESCRIPTION
**Pull request by sophiemachin**

### CODE REVIEW

File: src/user_registration.py [hunk 1]
```diff
@@ -18,44 +18,44 @@
...
+    user_data = event['details']['detail']['data']['details']['body']['user_metadata']
+    details['first_name'] = user_data['first_name']
+    details['last_name'] = user_data['last_name']
+    details['country_name'] = user_data['country']
+    details['id'] = user_data['citsci_uuid']
+
+    correlation_id = new_correlation_id()
+    hs_client = HubSpotClient(correlation_id=correlation_id)
+
+    notify_new_user_registration(details, correlation_id, stack_name='thiscovery-crm')
```
Comment (afs25) [inline src/user_registration.py:48]: The notification being saved to the Notifications table is rather large and unecessarily complex:

```
{
    'processing_status': 'new',
    'label': 'altha@email.co.uk',
    'created': '2022-03-10 19:04:00.267056+00:00',
    'details': {
        'created': '2022-02-25 10:48:43.852909+00:00',
        'resources': [],
        'last_name': 'Alcorn',
        'source': 'REDACTED',
        'version': '0',
        'detail-type': 'Auth0 log',
        'country_name': 'GB',
        'id': 'd1070e81-557e-40eb-a7ba-b951ddb7ebdc',
        'time': '2022-02-25T10:48:43Z',
        'detail': {
            'data': {
                'date': '2022-02-25T10:48:31.036Z',
                'log_id': '90020220225104831160441788900872658485194909784700092514',
                'user_name': 'altha@email.co.uk',
                'ip': 'REDACTED',
                'description': '',
                'strategy_type': 'database',
                'type': 'ss',
                'client_id': 'REDACTED',
                'connection_id': 'con_C72rtu1sAF9UFXmR',
                'user_id': 'REDACTED',
                'connection': 'Username-Password-Authentication',
                'details': {
                    'body': {
                        'password': '*****',
                        'user_metadata': {
                            'last_name': 'Alcorn',
                            'country': 'GB',
                            'first_name': 'Altha',
                            'citsci_uuid': 'd1070e81-557e-40eb-a7ba-b951ddb7ebdc'
                        },
                        'name': 'Altha Alcorn',
                        'connection': 'Username-Password-Authentication',
                        'tenant': 'thiscovery',
                        'client_id': 'REDACTED',
                        'email': 'altha@email.co.uk'
                    }
                },
                'strategy': 'auth0',
                'client_name': 'Thiscovery',
                'user_agent': 'GuzzleHttp/6.3.3 curl/7.58.0 PHP/7.4.26'
            },
            'log_id': '90020220225104831160441788900872658485194909784700092514'
        },
        'region': 'REDACTED',
        'first_name': 'Altha',
        'account': 'REDACTED',
        'email': 'altha@email.co.uk'
    },
    'id': 'd1070e81-557e-40eb-a7ba-b951ddb7ebdc',
    'modified': '2022-03-10 19:04:00.267056+00:00',
    'type': 'user-registration'
}
```

File: src/user_registration.py [hunk 2]
```diff
@@ -18,44 +18,44 @@
...
+    hubspot_id, is_new = hs_client.post_new_user_to_crm(details)
+    logger.info(
+        "process_user_registration: hubspot details",
+        extra={
+            "notification_id": str(notification_id),
+            "hubspot_id": str(hubspot_id),
+            "isNew": str(is_new),
+            "correlation_id": str(correlation_id),
+        },
+    )
```
Comment (afs25) [inline src/user_registration.py:59]: Delete this from here. This lambda should only add a new notification to the processing queue (Dynamodb table). Processing those notifications is the job of a different lambda. We could invoke that other lambda asynchronously here, but let's keep it simple and not do that for now.

**Review by afs25 — CHANGES_REQUESTED** [review]
See inline comments

File: src/user_registration.py (hunk 1 above)
Comment (afs25) [inline src/user_registration.py:48]: use common.constants.STACK_NAME here (line 48)?

**Comment by sophiemachin** [comment]
@afs25 I've fixed the issues you mentioned, but I also need to run the formatter. Considering it was a tricky last time, I think I'll let you check these changes then push again, if that works?

**Comment by afs25** [comment]
> @afs25 I've fixed the issues you mentioned, but I also need to run the formatter. Considering it was tricky last time to pick out actual changes vs formatting changes, I think I'll let you check these changes then push again after I've run it, if that works?

I like to set black as a FileWatcher in PyCharm so I don't have to worry about running it as a separate step. I've set that up following step 6 of https://black.readthedocs.io/en/stable/integrations/editors.html#pycharm-intellij-idea

**Review by afs25 — APPROVED** [review]

**Comment by sophiemachin** [comment]
I've set up the automatic save for black. For PyCharm I had to also update Actions on Save to Format code (in addition to running black)

**Pull request merged**